File: C++/05_Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int st=0;
        int en=nums.size()-1;
        while(st<=en)
        {
            int mid=(st+en)/2;
            int diff=nums[mid]-target;
            if(diff==0)
            {
                return mid;
            }
            else if(diff>0)
            {
                en--;
            }
            else st++;
        }
        return st;
    }
};

File: Python/05_Search Insert Position.py
# 35. Search Insert Position
# https://leetcode.com/problems/search-insert-position/

class Solution:
    def searchInsert(self, nums, target):
        start=0
        end=len(nums)-1
        while start<=end:
            mid=(start+end)//2
            if target>nums[mid]:
                start=mid+1
            else:
                end=mid-1
        return start